Mathlib/MeasureTheory/Integral/SetIntegral.lean

```diff
@@ -403,35 +403,23 @@ theorem set_integral_eq_integral_of_forall_compl_eq_zero (h : ∀ x, x ∉ s →
   set_integral_eq_integral_of_ae_compl_eq_zero (eventually_of_forall h)
 #align measure_theory.set_integral_eq_integral_of_forall_compl_eq_zero MeasureTheory.set_integral_eq_integral_of_forall_compl_eq_zero
 
+-- TODO: Do we need these 2 lemmas?
 lemma ae_restrict_eq_const_iff_ae_eq_const_of_mem {E : Type*} [MeasurableSpace E]
     [MeasurableSingletonClass E] {f : α → E} (c : E) {s : Set α}
     (f_mble : NullMeasurable f (μ.restrict s)) :
-    f =ᵐ[Measure.restrict μ s] (fun _ ↦ c) ↔ ∀ᵐ x ∂μ, x ∈ s → f x = c := by
-  simp only [Measure.ae, MeasurableSet.compl_iff, EventuallyEq, Filter.Eventually,
-             Pi.zero_apply, Filter.mem_mk, mem_setOf_eq]
-  rw [Measure.restrict_apply₀]
-  · constructor <;> intro h <;> rw [← h] <;> congr <;> ext x <;> aesop
-  · apply NullMeasurableSet.compl
-    convert f_mble (MeasurableSet.singleton c)
+    f =ᵐ[Measure.restrict μ s] (fun _ ↦ c) ↔ ∀ᵐ x ∂μ, x ∈ s → f x = c :=
+  ae_restrict_iff₀ <| by convert f_mble (measurableSet_singleton c) -- much faster with `convert`
 
 lemma ae_restrict_eq_const_iff_ae_eq_const_of_mem' {E : Type*} (c : E) (f : α → E) {s : Set α}
     (s_mble : MeasurableSet s) :
-    f =ᵐ[Measure.restrict μ s] (fun _ ↦ c) ↔ ∀ᵐ x ∂μ, x ∈ s → f x = c := by
-  simp only [Measure.ae, MeasurableSet.compl_iff, EventuallyEq, Filter.Eventually,
-             Pi.zero_apply, Filter.mem_mk, mem_setOf_eq]
-  rw [Measure.restrict_apply_eq_zero']
-  · constructor <;> intro h <;> rw [← h] <;> congr <;> ext x <;> aesop
-  · exact s_mble
+    f =ᵐ[Measure.restrict μ s] (fun _ ↦ c) ↔ ∀ᵐ x ∂μ, x ∈ s → f x = c :=
+  ae_restrict_iff' s_mble
 
 /-- If a function equals zero almost everywhere w.r.t. restriction of the measure to `sᶜ`, then its
 integral on `s` coincides with its integral on the whole space. -/
 lemma set_integral_eq_integral_of_ae_restrict_eq_zero (hs : f =ᵐ[μ.restrict sᶜ] 0) :
-    ∫ ω in s, f ω ∂μ = ∫ ω, f ω ∂μ := by
-  borelize E
-  refine set_integral_eq_integral_of_ae_compl_eq_zero ?_
-  have f_mble : NullMeasurable f (μ.restrict sᶜ) :=
-    NullMeasurable.congr measurable_const.nullMeasurable hs.symm
-  simpa only [mem_compl_iff] using (ae_restrict_eq_const_iff_ae_eq_const_of_mem 0 f_mble).mp hs
+    ∫ ω in s, f ω ∂μ = ∫ ω, f ω ∂μ :=
+  set_integral_eq_integral_of_ae_compl_eq_zero <| ae_imp_of_ae_restrict hs
```

Comment:
Do we need this lemma? Why not use `set_integral_eq_integral_of_ae_compl_eq_zero`? @kkytola

Reply:
I tried to check any uses, and I indeed now believe we don't need this.

Mathematically `set_integral_eq_integral_of_ae_compl_eq_zero` was clearly always the heart of this lemma. Probably I just struggled to get the hypotheses of that as a one-liner from the a.e. restriction to complement assumption that I had, and I added the lemma as an API (unnecessarily, as you show) in order not to clutter some actual proof. But it seems there are more straightforward ways, so it looks like a good idea to remove this.